lib/std/os/linux.zig

```diff
@@ -1785,32 +1791,48 @@ pub fn accept4(fd: i32, noalias addr: ?*sockaddr, noalias len: ?*socklen_t, flag
 pub fn fstat(fd: i32, stat_buf: *Stat) usize {
     if (@hasField(SYS, "fstat64")) {
         return syscall2(.fstat64, @as(usize, @bitCast(@as(isize, fd))), @intFromPtr(stat_buf));
-    } else {
+    } else if (@hasField(SYS, "fstat")) {
         return syscall2(.fstat, @as(usize, @bitCast(@as(isize, fd))), @intFromPtr(stat_buf));
+    } else {
+        // riscv32 has made the interesting decision to not implement some of
+        // the older stat syscalls, including this one.
+        @compileError("No fstat syscall on this architecture.");
```

Comment:
```suggestion
    } else if (native_arch == .riscv32) {
        @compileError("No fstat syscall on this 
    } else {
        return syscall2(.fstat, @as(usize, @bitCast(@as(isize, fd))), @intFromPtr(stat_buf));
```

Reply:
Do we really want to be in the business of manually checking each arch rather than just relying on the generated syscalls? For example, for `lstat`, we need to include arc too.

Reply:
yes we shouldnt throw this error for targets that simply arent implemented/havent been generated yet

Reply:
and yes this change should likely be done on all the syscall changes, just didnt wanna barage you with the same comment multiple times

Reply:
Is it any worse than the compile errors you'll get because `syscall2` and `SYS.fstat` aren't defined?

Reply:
yes because the explicit `@compileError` makes it look like the target was checked by hand and found not to have it